pkg/onmetal/create_machine_test.go

```diff
@@ -157,20 +155,24 @@ var _ = Describe("CreateMachine", func() {
 		Eventually(func(g Gomega) {
 			_, err := (*drv).CreateMachine(ctx, &driver.CreateMachineRequest{
 				Machine:      newMachine(ns, "machine", -1, nil),
-				MachineClass: newMachineClass("foo", internal.ProviderSpec),
+				MachineClass: newMachineClass("foo", testing.SampleProviderSpec),
 				Secret:       providerSecret,
 			})
 			g.Expect(err.Error()).To(ContainSubstring("not supported by the driver"))
 		}).Should(Succeed())
 
 		By("failing if the invalid IP is set for dnsServers in provider")
 		Eventually(func(g Gomega) {
+			testing.SampleProviderSpec["dnsServers"] = append(testing.SampleProviderSpec["dnsServers"].([]net.IP), net.ParseIP("invalid-dns-ips"))
```

Comment:
Can you create a local copy of the `SampleProviderSpec`. That way you don't need the defer call. One option would be to implement a getter function into the `testing` package.

Reply:
Done

Reply:
Done